server/src/main/java/io/druid/server/coordinator/CachingCostBalancerStrategy.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.coordinator;
+
+import com.google.common.base.Preconditions;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import io.druid.server.coordinator.cost.ClusterCostCache;
+import io.druid.timeline.DataSegment;
+
+
+public class CachingCostBalancerStrategy extends CostBalancerStrategy
```

Comment:
I think the only reason to introduce another configuration option is that `CachingCostBalancerStrategy` has bigger memory footprint that `CostBalancerStrategy`. Not sure that all users are ok with such change